lib/cagen.py

```diff
@@ -181,7 +181,7 @@ def voms(self, vo_name):
             if exc.errno == errno.EEXIST:
                 pass
 
-        uri = vo_name + '.opensciencegrid.org'
+        uri = _get_hostname()
```

Comment:
This is gonna blow up if `_get_hostname()` returns None.

Reply:
Constructing the host cert will break, too, which would really ruin people's day. Should I just not catch `socket.error` in `_get_hostname()` and let it blow up?

Reply:
Expected problems should blow up nicely so that if it happens, we don't think it's a bug. Wrap it in an exception of our own or something.